dashboard/test/ui/features/star_labs/applab/data_tab.feature

```diff
@@ -0,0 +1,67 @@
+# Maddie (10/19/2020) disabled in IE because "And I press keys" step does not work in IE.
+# https://github.com/code-dot-org/code-dot-org/pull/24646
+@no_ie
+@no_mobile
+@as_student
+Feature: App Lab Data Tab
+
+  Background:
+    # Navigate to data tab from a new Applab project
+    Given I start a new Applab project
+    And I wait for the page to fully load
+    Then I click selector "#dataModeButton"
+    And I wait until element "#dataTablesBody" is visible
+
+  Scenario: Datasets Panel
+    # Import a dataset
+    Then I click selector ".uitest-dataset-category:first-of-type" once I see it
+    Then I click selector ".uitest-dataset-table-link:first-of-type" once I see it
+    Then I click selector ".uitest-dataset-preview-btn:first-of-type" once I see it
+    And I wait until element ".modal h1" is visible
+    Then I save the table name from element ".modal h1"
+    Then I click selector ".modal button:first-of-type" once I see it
+    Then I wait until element "#dataTablesBody table tr:eq(2) td:eq(0)" contains the saved table name
+
+  Scenario: Data Tables Tab
+    # Create a new table
+    And I press keys "My new table" for element ".uitest-add-table-input:first-of-type"
+    And I click selector ".uitest-add-table-btn:first-of-type"
+    Then I wait until element "#dataTable span:first-of-type" contains text "My new table"
+
+    # Rename default column
+    And I wait until element ".test-tableNameDiv:first-of-type" contains text "column1"
+    And I click selector ".dropdown.pull-right:first-of-type"
+    And I press keys "number column" for element ".uitest-data-table-content:first-of-type th:nth-child(2) input"
+    When I press enter key
+    Then I wait until element ".test-tableNameDiv:first-of-type" contains text "number column"
+
+    # Add a row
+    And I click selector "#addDataTableRow input:first-of-type"
+    And I press keys "2" for element "#addDataTableRow input:first-of-type"
+    And I click selector "#addDataTableRow button:first-of-type"
+    Then I wait until element ".uitest-data-table-content:first-of-type tr:nth-child(3) td:nth-child(2)" contains text "2"
```

Comment:
Is it worth adding a unique identifier for some of these more complex selectors?  Not sure whether this selector might be a bit fragile long term...

Reply:
Alternatively, maybe these table-specific lines could be factored out into a helper?

Reply:
yeah, especially coming back to this PR after putting it down for a while, these are difficult to read. i'll start with adding classes to the row (and other) elements that have lengthy selectors